src/core/congestion_control.c

```diff
@@ -538,3 +547,27 @@ QuicCongestionControlOnDataLost(
     QuicCongestionControlUpdateBlockedState(Cc, PreviousCanSendState);
     QuicConnLogCubic(QuicCongestionControlGetConnection(Cc));
 }
+
+_IRQL_requires_max_(DISPATCH_LEVEL)
+void
+QuicCongestionControlOnSpuriousCongestionEvent(
+    _In_ QUIC_CONGESTION_CONTROL* Cc
+    )
+{
+    BOOLEAN PreviousCanSendState = QuicCongestionControlCanSend(Cc);
+
+    //
+    // Revert to previous state.
+    //
+    Cc->WindowMax = Cc->PrevWindowMax;
```

Comment:
How are we guaranteed at this point that the PrevXxx variables were initialized? It seems pretty indirect: we had a nonempty LostPackets list, but it's not the QuicCongestionControlOnCongestionEvent function that puts packets in that list. And I see QuicCongestionControlOnCongestionEvent isn't even always called by QuicCongestionControlOnDataLost, which in turn is called by a function (QuicLossDetectionDetectAndHandleLostPackets) that doesn't add packets to that list. I.e. this assumption is very complicated.

Reply:
I added a return at the beginning if somehow we weren't in recovery.